src/codegen/struct_layout.rs

```diff
@@ -236,11 +239,14 @@ impl<'a> StructLayoutTracker<'a> {
                 field_layout
             );
 
+            let padding_align = if force_padding {
+                0
```

Comment:
Using an alignment of zero seems wrong, shouldn't it be one?

Reply:
Yes, sorry. Will fix.

Reply:
Actually, thinking a bit more about it, shouldn't the padding alignment be the default if it's mandatory padding? Otherwise we might not fulfill the alignment requirements.

Reply:
I'm afraid I don't understand. The padding I see emitted is an array of `u8`s, e.g.
```c
struct pad_me {
        uint8_t first;
        uint32_t second;
        uint16_t third;
};
```
```rust
pub struct pad_me {
    pub first: u8,
    pub __bindgen_padding_0: [u8; 3usize],
    pub second: u32,
    pub third: u16,
    pub __bindgen_padding_1: [u8; 2usize],
}
```

I don't understand why any `__bindgen_padding_X` field would ever have an alignment requirement-- that sounds like something that would re-introduce gaps between fields, which is exactly what the padding is meant to eliminate.

Reply:
I may be confused by the fact that I don't actually know what the existing padding code is for!

My desire is to get a struct that has no gaps-- every byte is accounted for by a named field.  The previous code is kind of a mystery to me; I don't know what problem it's solving.